libs/libc/signal/sig_ismember.c

```diff
@@ -63,17 +66,53 @@
  *
  ****************************************************************************/
 
-int sigismember(FAR const sigset_t *set, int signo)
+int nxsig_ismember(FAR const sigset_t *set, int signo)
 {
-  int ret = ERROR;
-
   /* Verify the signal */
 
-  if (GOOD_SIGNO(signo))
+  if (!GOOD_SIGNO(signo))
+    {
+      return ERROR;
+    }
+  else
     {
       /* Check if the signal is in the set */
 
-      ret = ((*set & SIGNO2SET(signo)) != 0);
+      return ((*set & SIGNO2SET(signo)) != 0);
+    }
+}
+
+/****************************************************************************
+ * Name: sigismember
+ *
+ * Description:
+ *   This function tests whether the signal specified by signo is a member
+ *   of the set specified by set.
+ *
+ * Input Parameters:
+ *   set - Signal set to test
+ *   signo - Signal to test for
+ *
+ * Returned Value:
+ *   1 (true), if the specified signal is a member of the set,
+ *   0 (OK or FALSE), if it is not, or
+ *  -1 (ERROR) if the signal number is invalid.
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+int sigismember(FAR const sigset_t *set, int signo)
+{
+  int ret;
+
+  /* Let nxsig_ismember do all of the work */
+
+  ret = nxsig_ismember(set, signo);
+  if (ret < 0)
+    {
+      set_errno(EINVAL);
```

Comment:
In the general case, it should set_errno(-ret); In this case, -EINVAL should be the only value returned by nxsig_ismember() so it really doesn't matter